From d8190aae3ff2f0a63a8fb72922fb117b5ab34b6a Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@redhat.com>
Date: Tue, 28 Mar 2023 19:42:53 -0500
Subject: [PATCH] wayland: Don't crash without xdg_activation_v1

This backports 3cdb6392102079d85ea38fa24d58c1824fe4dd37 to GTK 3.

See: #5701
---
 gdk/wayland/gdkwindow-wayland.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index 619f7982cbe..9c0264e71b6 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -4179,7 +4179,7 @@ gdk_wayland_window_set_startup_id (GdkWindow   *window,
     }
 
 #ifdef HAVE_XDG_ACTIVATION
-  if (startup_id)
+  if (display_wayland->xdg_activation && startup_id)
     {
       xdg_activation_v1_activate (display_wayland->xdg_activation,
                                   startup_id,
-- 
GitLab

